prow/scripts/cluster-integration/compass-gke-integration.sh

```diff
@@ -158,7 +153,8 @@ function createCluster() {
     export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
   fi
   CLEANUP_CLUSTER="true"
-  env ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}" "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}"/provision-gke-cluster.sh
+  export ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}"
+  gcloud::provision_gke_cluster "$CLUSTER_NAME"
```

Comment:
This seems to hide functionality, couldn't we have this more explizit by doing something like this instead?
```sh
gcloud::provision_gke_cluster "${CLUSTER_NAME}" "${ADDITIONAL_LABELS}"
```

Reply:
There is problem with optional arguments: adding something like optional `$1` or `$@` in a function body and not adding it to any of the function call (e.g. just calling it by `cloud::provision_gke_cluster "${CLUSTER_NAME}"`) will make shellcheck complain and fail. Probably only way do do it it to disable one of the shellchecks for the whole function:
https://github.com/koalaman/shellcheck/wiki/SC2120

Reply:
I'm thinking more of a change like this:
```sh
function gcloud::provision_gke_cluster {
  if [ -z "$1" ]; then
    log::error "Cluster name not provided. Provide proper cluster name."
    exit 1
  fi
  CLUSTER_NAME=$1
  if [ -z "$2" ]; then
    log::info "No additional labels provided"
  fi
  ADDITIONAL_LABELS=$2
```

This way you can still use it like this: `cloud::provision_gke_cluster "${CLUSTER_NAME}"` and from my a quick test shellcheck would not complain here, unless I'm missing something?

open https://www.shellcheck.net/, use this code:
```sh
#!/usr/bin/env bash

function gcloud::provision_gke_cluster {
  if [ -z "$1" ]; then
    log::error "Cluster name not provided. Provide proper cluster name."
    exit 1
  fi
  CLUSTER_NAME=$1
  if [ -z "$2" ]; then
    log::info "No additional labels provided"
  fi
  ADDITIONAL_LABELS=$2
  echo "${CLUSTER_NAME}"
  echo "${ADDITIONAL_LABELS}"
}

gcloud::provision_gke_cluster "hello"

gcloud::provision_gke_cluster "hello" "world"
```

Also the comment on the function does not indicate that providing `ADDITIONAL_LABELS` will have influence on anything that this function creates.